### DISCUSSION TITLE
Why does the checkbox not support the 'required' prop?

### DISCUSSION THREAD

**Discussion by charliematters** [tailwindlabs/headlessui#3510] [Help]
I'm a big fan of the new catalyst framework, but I've been tripped up by the checkbox component not supporting the 'required' attribute detailed here: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#validation

Is this an oversight, or a deliberate decision. If it's the latter, is there an example of how I can get it to play nice with the HTML5 form validity states?

**Comment by j3sch** [comment]
I don't understand either. RadixUI have it: https://www.radix-ui.com/primitives/docs/components/checkbox#api-reference

**Comment by ivan-kleshnin** [comment]
> I'm a big fan of the new catalyst framework, but I've been tripped up by the checkbox component not supporting the 'required' attribute detailed here: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#validation

Catalyst relies on Headless UI where `Checkbox` is implented as `span`. Spans have no notion of `required`. 
So the question should be phrased as "why Headless UI does not use native `<input type="checkbox">` element". 

Personally – I don't know, these guys rarely explain their design decisions.

Unlike `Input`, `Checkbox` also does not support `invalid`. And there're a lot of gotchas like that.

**Reply by arshad-yaseen** [reply]
Yeah, very correct.
